source/common/stats/thread_local_store.h

```diff
@@ -272,6 +272,24 @@ class ThreadLocalStoreImpl : Logger::Loggable<Logger::Id::stats>, public StoreRo
     CentralCacheEntry central_cache_;
   };
 
+  struct NullScopeImpl : public ScopeImpl {
```

Comment:
Won't this still mean that the TLS maps get populated with a bunch of stats (even if they're the same stat, you'll still be storing all their names as the keys, right)?  If you want to avoid this, I would suggest overriding the `counter()`, `gauge()`, and `histogram()` methods to just directly return the same stat each time without notifying TLS.  You also wouldn't need to replace the allocator because you can just store the `Counter`, `Gauge`, and `Histogram` in this class rather than asking the allocator to create it for you.